#! /bin/sh /usr/share/dpatch/dpatch-run
## build-system.dpatch by  <tim@retout.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Changes to f-spot build system from Debian package version 0.3.5-1.2

@DPATCH@

commit 262c969b3e1fa0f4085c9e447752dfa10180bfb8
Author: Tim Retout <tim@retout.co.uk>
Date:   Sat Dec 8 20:40:37 2007 +0000

    Imported Debian patch 0.3.5-1.2

diff --git a/Makefile.am b/Makefile.am
index 2e8c1ca..8dca427 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,11 @@
+if !EXTERNAL_DBUS
+DBUS_DIRS =             \
+	dbus-sharp      \
+	dbus-sharp-glib 
+endif
+
 SUBDIRS = 		\
-	dbus-sharp	\
-	dbus-sharp-glib	\
+	$(DBUS_DIRS)    \
 	docs		\
 	FlickrNet	\
 	SmugMugNet	\
diff --git a/libeog/Makefile.am b/libeog/Makefile.am
index 4152d1f..aed4a98 100644
--- a/libeog/Makefile.am
+++ b/libeog/Makefile.am
@@ -44,6 +44,7 @@ libfspoteog_files =				\
 
 libfspoteog_la_SOURCES = $(libfspoteog_files)
 
+libfspoteog_la_LIBADD = $(F_LIBS) $(LCMS_LIBS)
 
 # GLib marshalling cruft
 
diff --git a/libfspot/Makefile.am b/libfspot/Makefile.am
index d37c522..df39e01 100644
--- a/libfspot/Makefile.am
+++ b/libfspot/Makefile.am
@@ -30,7 +30,8 @@ libfspot_la_SOURCES =				\
 	f-utils.h
 
 libfspot_la_LIBADD = 					\
-	$(LCMS_LIBS)					\
+	$(F_LIBS) \
+	$(LCMS_LIBS) \
 	$(EXIF_LIBS)					\
 	$(top_builddir)/libjpegtran/libfspotjpegtran.la	\
 	$(top_builddir)/libeog/libfspoteog.la
diff --git a/libjpegtran/Makefile.am b/libjpegtran/Makefile.am
index 96465d0..b2f0d1a 100644
--- a/libjpegtran/Makefile.am
+++ b/libjpegtran/Makefile.am
@@ -18,5 +18,6 @@ libfspotjpegtran_files =			\
 libfspotjpegtran_la_SOURCES =			\
 	$(libfspotjpegtran_files)
 
+libfspotjpegtran_la_LIBADD = $(LIBJPEG) $(F_LIBS)
 
 EXTRA_DIST = README
commit 262c969b3e1fa0f4085c9e447752dfa10180bfb8
Author: Tim Retout <tim@retout.co.uk>
Date:   Sat Dec 8 20:40:37 2007 +0000

    Imported Debian patch 0.3.5-1.2

diff --git a/configure.in b/configure.in
index f6915f7..2efcd00 100644
--- a/configure.in
+++ b/configure.in
@@ -104,6 +104,16 @@ dnl --- GConf
 AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
 AM_GCONF_SOURCE_2
 
+dnl --- NDesk.DBus
+
+PKG_CHECK_MODULES(DBUS, ndesk-dbus-1.0 >= 0.4 \
+	ndesk-dbus-glib-1.0 >= 0.3, have_external_dbus="yes", have_external_dbus="no")
+if test "x$have_external_dbus" != "xyes"; then
+	AC_MSG_RESULT([using internal copy])
+fi
+AM_CONDITIONAL(EXTERNAL_DBUS, test "x$have_external_dbus" = "xyes")
+AC_SUBST(DBUS_LIBS)
+
 
 dnl --- libjpeg
 
@@ -163,6 +173,11 @@ fi
 
 GNOME_SCREENSAVER_THEMESDIR=$gnome_screensaver_prefix/share/gnome-screensaver/themes
 
+if pkg-config --exists gnome-screensaver ; then
+  GNOME_SCREENSAVER_SAVERDIR=$(pkg-config --variable=privlibexecdir gnome-screensaver)
+  GNOME_SCREENSAVER_THEMESDIR=$(pkg-config --variable=themesdir gnome-screensaver)
+fi
+
 AC_SUBST(GNOME_SCREENSAVER_SAVERDIR)
 AC_SUBST(GNOME_SCREENSAVER_THEMESDIR)
 
commit 262c969b3e1fa0f4085c9e447752dfa10180bfb8
Author: Tim Retout <tim@retout.co.uk>
Date:   Sat Dec 8 20:40:37 2007 +0000

    Imported Debian patch 0.3.5-1.2

diff --git a/Makefile.include b/Makefile.include
index 815cc9f..987e14d 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -24,9 +24,13 @@ DIR_TAO_EXTENSIONLOADER = $(top_builddir)/Tao/Tao.OpenGl.ExtensionLoader
 DIR_TAO_GLPOSTPROCESS = $(top_builddir)/Tao/Tao.GlPostProcess
 ## Links
 
+if EXTERNAL_DBUS
+LINK_DBUS = $(DBUS_LIBS)
+else
 LINK_DBUS = 					\
 	-r:$(DIR_DBUS)/NDesk.DBus.dll		\
-	-r:$(DIR_DBUS_GLIB)/NDesk.DBus.GLib.dll
+	-r:$(DIR_DBUS)/NDesk.DBus.GLib.dll
+endif
 LINK_FLICKR = -r:$(DIR_FLICKR)/FlickrNet.dll
 LINK_KEYRING = -r:$(DIR_KEYRING)/gnome-keyring-sharp.dll
 LINK_GLITZ = -r:$(DIR_GLITZ)/NDesk.Glitz.dll 
